File: 백준/Silver/11068. 회문인 수/회문인 수.py
t = int(input())

def solution(n: int):
    # 목표: 2 ~ 64 진법으로 변경되었을 때 회문이 될 수 있는가?

    # 1. 10진법 n -> m(2 ~ 64)진법으로 변경
    def conv(n: int, m: int) -> str:
        if n == 0:
            return 0
        # 10 + 26 + 26 + 2 = 64
        conv = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!@"
        ans = ""
        while n > 0:
            n, mod = divmod(n, m)
            ans += conv[mod]
        return ans[::-1]
    
    # 2. 회문인지 파악
    def is_palindrom(x: str):
        k = len(x)
        for i in range(k // 2):
            if x[i] != x[k - i- 1]:
                return False
        return True

    flag = False
    for m in range(2, 65):
        res = conv(n, m)
        if is_palindrom(res):
            flag = True
    
    if flag:
        return 1
    return 0

for _ in range(t):
    x = int(input())
    print(solution(x))

File: 백준/Silver/11068. 회문인 수/회문인 수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        for (int i = 0; i < t; i++) {
            int number = Integer.parseInt(br.readLine());
            System.out.println(check(number));
        }
    }

    private static int check(int number) {
        int ans = 0;
        for (int m = 2; m < 65; m++) {
            String res = convert(number, m);
            if (isPalindrome(res)) {
                ans = 1;
            }
        }
        return ans;
    }

    // 회문 확인하기
    private static boolean isPalindrome(String str) {
        for (int idx = 0; idx < str.length() / 2; idx++) {
            if (str.charAt(idx) != str.charAt(str.length() - idx - 1)) {
                return false;
            }
        }
        return true;
    }

    // 10진법 number -> m진법으로 변경
    private static String convert(int number, int m) {
        // reverse를 사용하기 위해서 StringBuilder 사용!
        StringBuilder res = new StringBuilder();
        // 0 - 9 : 10
        // a - z : 26
        // A - z : 26
        // @, * : 2
        // 64
        String conv = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ@*";

        if (number == 0) {
            res.append("0");
            return "0";
        }

        while (number > 0) {
            // String.valueOf: object -> String
            String k = String.valueOf(conv.charAt(number % m));
            res.append(k);
            number /= m;
        }
        // StringBuilder -> String
        return res.reverse().toString();
    }
}